workbench/icons/Makefile.am

```diff
@@ -0,0 +1,20 @@
+icondir = $(datadir)/icons/hicolor/16x16/apps
```

Comment:
The icons are obtained from GTK, so they are not in Geany, [see](https://github.com/geany/geany/blob/adc22a453b546b05f507fd6522ea13f3a92943c2/src/ui_utils.c#L3017).  And IIUC GTK gets them from the theme these days.

My Geany shows a JS icon in the documents tab for js files, so maybe its your theme thats deficient.